flycheck.el

```diff
@@ -8456,6 +8457,42 @@ See URL `http://www.gnu.org/software/texinfo/'."
           line-end))
   :modes texinfo-mode)
 
+(flycheck-def-config-file-var flycheck-typescript-tslint-config typescript-tslint "tslint.json"
+  :safe #'stringp
+  :package-version '(flycheck . "0.26"))
+
+(flycheck-def-option-var flycheck-typescript-tslint-rulesdir nil typescript-tslint
+  "The directory of custom rules for TSLint.
+
+The value of this variable is either a string containing the path
+to a directory with custom rules, or nil, to not give any custom
+rules to TSLint.
+
+Refer to the TSLint manual at URL
+`http://palantir.github.io/tslint/usage/cli/'
+for more information about the custom directory."
+  :type '(choice (const :tag "No custom rules directory" nil)
+                 (directory :tag "Custom rules directory"))
+  :safe #'stringp
+  :package-version '(flycheck . "0.26"))
+
+(flycheck-define-checker typescript-tslint
+  "TypeScript syntax and style checker using TSLint.
+
+Note that this syntax checker is not used if
+`flycheck-typescript-tslint-config' is nil or refers to a non-existing file.
+
+See URL
+`https://github.com/palantir/tslint'."
+  :command ("tslint"
+            "--format" "prose"
+            (config-file "--config" flycheck-typescript-tslint-config)
+            (option "--rules-dir" flycheck-typescript-tslint-rulesdir)
+            source)
+  :error-patterns
+  ((warning line-start (one-or-more not-newline) "[" line ", " column "]: " (message) line-end))
```

Comment:
What's in the `(one-or-more not-newline)` group?

By the way, I noticed that tslint can output JSON.  Would it be possible to write a custom `:error-parser` for this JSON format?  Generally, we prefer to parse structured output formats from syntax checkers instead of the human-readable error output, because in our experience the structured formats are a lot less fragile and don't change very often, which makes maintenance much easier for us on the long run.